pkg/testworkflows/testworkflowexecutor/executor.go

```diff
@@ -467,7 +468,23 @@ func (e *executor) initialize(ctx context.Context, workflow *testworkflowsv1.Tes
 		RunningContext:            request.RunningContext,
 	}
 
-	// Try to resolve tags initially
+	// Store the configuration if it is small and not sensitive
+	if request.Config != nil && testworkflows.CountMapBytes(request.Config) < 10*1024*1024 {
+		storeConfig := true
+		schema := workflow.Spec.Config
+
+		for k, _ := range request.Config {
+			if s, ok := schema[k]; ok && s.Sensitive {
+				storeConfig = false
+			}
+		}
```

Comment:
I'm not 100% sure, but could be that iterating over `workflow.Spec.Config` could be more correct - as a sensitive property set to default value may also be kind of sensitive *(similar to uncertainty when working with PII rules - single information may not be sensitive, but in conjunction with others - it can become sensitive)*

But, I guess, we can ignore that for now 👍

Reply:
> I.e. if it's kind of sensitive in relation to others, you can still set it to sensitive, right?

Yes, but I'm more saying about the default value - i.e. if you specify:

```yaml
spec:
  config:
    apiUrl: { type: string, default: "my-value", sensitve: true }
```

And you request execution with config:
```json
{}
```

The code above will save the configuration properly, despite it actually is using sensitive `my-value` value.

As mentioned though, I think that we can ignore it for now :)